WordPress/Classes/ViewRelated/Plugins/PluginDirectoryViewModel.swift

```diff
@@ -34,73 +38,234 @@ class PluginDirectoryViewModel: Observable {
+                                              listViewQuery: query,
+                                              noResultsView: noResults,
+                                              presenter: presenter)
+        }
+        return nil
+    }
+
+    private func newRow(presenter: PluginPresenter & PluginListPresenter) -> ImmuTableRow? {
+        let title = NSLocalizedString("New", comment: "Header of section in Plugin Directory showing newest plugins")
+        let secondaryTitle = NSLocalizedString("See All", comment: "Button in Plugin Directory letting users see more plugins")
+        let query = PluginQuery.feed(type: .newest)
+
+        if let new = newPlugins {
+            return CollectionViewContainerRow(title: title,
+                                                secondaryTitle: secondaryTitle,
+                                                plugins: new,
                                                 site: site,
                                                 accessoryViewCallback: accessoryViewCallback,
-                                                listViewQuery: .feed(type: .newest),
+                                                listViewQuery: query,
                                                 presenter: presenter)
+        } else if let noResults = noResultsView(for: query) {
+            return CollectionViewContainerRow(title: title,
+                                              secondaryTitle: secondaryTitle,
+                                              site: site,
+                                              listViewQuery: query,
+                                              noResultsView: noResults,
+                                              presenter: presenter)
+        }
+
+        return nil
+    }
 
-        return ImmuTable(sections: [
-            ImmuTableSection(rows: [
-                installedRow,
-                featuredRow,
-                popularRow,
-                newRow,
+    private var accessoryViewCallback: ((PluginDirectoryEntry) -> UIView) {
+        // We need to be able to map between a `PluginDirectoryEntry` and a potentially already installed `Plugin`,
+        // but passing the entire `PluginStore` to the CollectionViewContainerRow isn't the best idea.
+        // Instead, we allow the row to reach back to us and ask for the accessoryView.
+        return { [weak self] (entry: PluginDirectoryEntry) -> UIView in
+            guard let strongSelf = self else {
+                return UIView()
+            }
+            return strongSelf.accessoryView(for: entry)
+        }
+    }
+
+    func tableViewModel(presenter: PluginPresenter & PluginListPresenter) -> ImmuTable {
+        return ImmuTable(optionalSections: [
+            ImmuTableSection(optionalRows: [
+                installedRow(presenter: presenter),
+                featuredRow(presenter: presenter),
+                popularRow(presenter: presenter),
+                newRow(presenter: presenter),
```

Comment:
If the four sections are always displayed now, even on initial load, why is it possible for them to be optional?